crates/cast/src/cmd/call.rs

```diff
@@ -319,12 +320,18 @@ impl CallArgs {
             return Ok(());
         }
 
-        sh_println!(
-            "{}",
-            Cast::new(provider)
-                .call(&tx, func.as_ref(), block, state_overrides, block_overrides)
-                .await?
-        )?;
+        let response = Cast::new(&provider)
+            .call(&tx, func.as_ref(), block, state_overrides, block_overrides)
+            .await?;
+
+        if response == "0x" {
+            let contract_address = tx.to.unwrap().into_to().unwrap();
```

Comment:
unsure if this always holds, so I'd suggest putting this behind an `if let Some(contract_address) = tx.to.and_then(|kind| kind.into_tx){}`

Reply:
Done